File: Week_05/Seminar/Solutions/QueueAndStack/MaximumNestingDepthoftheParentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        int i = 0;
        int max = 0;
        for(auto ch: s) {
            if(ch == '(') {
                i++;
                if(i > max) {
                    max = i;
                }
            }
            else if(ch == ')') {
                i--;
            }
        }
        
        return max;
    }
};

File: Week_05/Practicum/python/Maximum Nesting Depth of the Parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        stack_ = []

        ans = 0
        
        for c in s:
            if c == '(':
                stack_.append(1)
                ans = max(ans, len(stack_))
            elif c == ')':
                stack_.pop()

        return ans